lib/Controller/EndpointController.php

```diff
@@ -214,4 +220,19 @@ protected function getCurrentUser() {
 
 		return (string) $user;
 	}
+
+	/**
+	 * @param string $link
+	 * @return string
+	 */
+	protected function getAbsoluteLink(string $link) {
+		$urlComponents = \parse_url($link);
+
+		// Check if already absolute
+		if (!isset($urlComponents['host'])) {
```

Comment:
is that safe? What if the hostname has changed and we still have old activities in the DB?